sdk/storagecache/Microsoft.Azure.Management.StorageCache/tests/Helpers/CacheHelper.cs

```diff
@@ -127,7 +133,34 @@ public Cache Create(string name, string sku, int cacheSize, bool skipGet = false
             {
                 var cacheSku = new CacheSku() { Name = sku };
                 var subnetUri = $"/subscriptions/{this.subscriptionId}/resourcegroups/{this.resourceGroup.Name}/providers/Microsoft.Network/virtualNetworks/{this.virtualNetwork.Name}/subnets/{this.subNet.Name}";
-                var cacheParameters = new Cache() { CacheSizeGB = cacheSize, Location = this.resourceGroup.Location, Sku = cacheSku, Subnet = subnetUri };
+                if (encryptionKeyURL is null || keyVaultResourceId is null)
+                {
+                    keyVaultKeyReference = new KeyVaultKeyReference() { };
+                    cacheEncryptionSettings = new CacheEncryptionSettings() { };
+                }
+                else
+                {
+                    keyVaultKeyReference = new KeyVaultKeyReference()
+                    {
+                        KeyUrl = encryptionKeyURL,
+                        SourceVault = keyVaultResourceId,
+                    };
```

Comment:
If encyrptionKeyURL or keyVaultResourceId is null, but not both, then either KeyUrl or SourceVault will be null, depending. If this is expected as part of testing, no-op.

Reply:
I thought both are required to make encryption work so i just check for one of the variable to be null, if so then do the default encryption settings that is "no encryption". 
Also i am planning to actually use key vault in the dotnet pipeline so we can have it tested there regularly.